### PR TITLE
Update review guidelines

### PR DESCRIPTION
**Pull request by jougs**
This PR updates the review guidelines based on observations by @terhorstd and me during the release making process. This should also be discussed regularly during hackathons.

### CODE REVIEW

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:39]: Before merging, reviewers have to make sure that:
```

Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:39]:
```suggestion
Before merging, reviewers have to make sure that:
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:46]:
```suggestion
   labeled
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:41]:
```suggestion
1. pull request titles match the actual content of the PR and
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:44]:
```suggestion
  upper-case, present tense verb and end without punctuation
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:16]:
```suggestion
* In most cases, each pull request needs an OK from the
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:24]:
```suggestion
  review can be waived, and a single review plus the OK from the CI
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:25]:
```suggestion
  system is sufficient to merge the request.
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:27]:
```suggestion
  need to be accompanied by one or more tests written in either SLI or
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:29]:
```suggestion
* Each change to the code has to be reflected in the
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:30]:
```suggestion
  corresponding examples and documentation, as well.
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:33]:
```suggestion
* Please check the **typeset documentation** as part of the review
```

Do you mean the cpp guidelines, and reStructuredText etc?

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst:21]:
```suggestion
  actual issue).
```

**Review by jessica-mitchell — CHANGES_REQUESTED** [review]
Just some minor suggestions

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jougs) [reply doc/userdoc/contribute/code_review_guidelines.rst:33]: I guess. This sentence has already been there before...

Comment (jessica-mitchell) [reply doc/userdoc/contribute/code_review_guidelines.rst:33]: Maybe it would be better to link to the contribute/index page? and to check relevant guidelines found there?

Comment (jougs) [reply doc/userdoc/contribute/code_review_guidelines.rst:33]: I somehow fail to see that connection at the moment. Could you please elaborate? For me, this point is about the fact that reviewers have to check the rendered version of documentation (if that applies) and the existing link to the documentation workflow seems appropriate to me.

Comment (jessica-mitchell) [reply doc/userdoc/contribute/code_review_guidelines.rst:33]: Ok I see, I misunderstood what was meant by typeset documentation. 
How about:
```suggestion
* Please check that the typeset in the output documentation looks correct as part of the review
```

Comment (jougs) [reply doc/userdoc/contribute/code_review_guidelines.rst:33]: Maybe like this?
```suggestion
* Please also check that the typesetting of the documentation looks correct.
```

Comment (jessica-mitchell) [reply doc/userdoc/contribute/code_review_guidelines.rst:33]: :+1:

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jougs) [inline doc/userdoc/contribute/code_review_guidelines.rst:34]:
```suggestion
* Please also check that the typesetting of the documentation looks
  correct. To learn how to test the documentation locally offline,
```

Comment (jessica-mitchell) [inline doc/userdoc/contribute/code_review_guidelines.rst]:
```suggestion
To learn how to test the documentation locally offline,
```

**Review by jessica-mitchell — APPROVED** [review]
There is one small mistake I commented on, but looks good otherwise. Approving. Thanks!

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (jougs) [reply doc/userdoc/contribute/code_review_guidelines.rst]: I've already addressed this.

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (heplesser) [inline doc/userdoc/contribute/code_review_guidelines.rst:19]:
```suggestion
  does the right thing and is architecturally sound) and the content
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (heplesser) [inline doc/userdoc/contribute/code_review_guidelines.rst:30]:
```suggestion
  corresponding examples and documentation.
```

File: doc/userdoc/contribute/code_review_guidelines.rst
Comment (heplesser) [inline doc/userdoc/contribute/code_review_guidelines.rst:44]:
```suggestion
  upper-case, present-tense verb and end without punctuation
```

**Review by heplesser — CHANGES_REQUESTED** [review]
I just have some small textual suggestions.

**Comment by jougs** [comment]
@heplesser: I've merged your suggestions. Can you please re-review/merge? Thanks!

**Review by heplesser — APPROVED** [review]

**Pull request merged**
